### ISSUE TITLE
Eval-only AMP

### ISSUE THREAD

**Issue by bradezard131** [PyTorchLightning/pytorch-lightning#2443]
## 🚀 Feature
A trainer flag to enable Automatic Mixed Precision only during Validation/Testing.

### Motivation

Some models cannot be trained with AMP because of peculiarities in their loss functions (e.g. Weakly Supervised methods based on WSDDN, which require the use of `F.binary_cross_entropy` which does not work under AMP). This doesn't prevent such models from running under AMP during evaluation though since the incompatible functions are in the loss. Especially for validation where the most precise accuracy metrics may not be necessary, this would be very useful.

### Pitch

Add a new flag (or flags) on `Trainer` along the lines of `use_amp_val`.

**Comment by Borda** [comment]
cc: @williamFalcon

**Comment by rohitgr7** [comment]
is this still an issue??
